src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -858,17 +858,118 @@ else if (projectsString.indexOf(siteId) == -1) {
+					+ remoteUser + " and site=" + siteId);
+			return rv;
+		}
+		
+		// continue if there is a migrationId param
+		HashMap<String, Object> sessionAttributes = Utils
+				.login_becomeuser(env, request, remoteUser);
+
+		StringBuffer boxMigrationErrors = new StringBuffer();
+		
+		// call asynchronous method for Box file upload
+		log.info("start to call Box migration asynch for siteId=" + siteId);
+
+		// need to create all folders first
+		// get box client id and secret
+		String boxClientId = env.getProperty(Utils.BOX_CLIENT_ID);
+		String boxClientSecret = env.getProperty(Utils.BOX_CLIENT_SECRET);
+		String boxClientRedirectUrl = env.getProperty(Utils.BOX_CLIENT_REDIRECT_URL);
+		String boxAPIUrl = env.getProperty(Utils.BOX_API_URL);
+		// need to have all Box app configurations
+		if (boxClientId == null || boxClientSecret == null) {
+			String boxClientIdError = "Missing Box integration parameters (Box client id, client secret)";
+			log.error(boxClientIdError);
+			boxMigrationErrors.append(boxClientIdError + Utils.LINE_BREAK);
+		}
+		
+		String remoteUserEmail = getUserEmailFromUserId(remoteUser);
+
+		if (siteId == null || boxFolderId == null) {
+			String boxFolderIdError = "Missing params for CTools site id, or target Box folder id.";
+			log.error(boxFolderIdError);
+			boxMigrationErrors.append(boxFolderIdError + Utils.LINE_BREAK);
+		}
+
+		// get sessionId
+		if (sessionAttributes.containsKey(Utils.SESSION_ID)) {
+			String sessionId = (String) sessionAttributes.get(Utils.SESSION_ID);
+			HttpContext httpContext = (HttpContext) sessionAttributes
+					.get("httpContext");
+
+			// 3. get the site resource list
+			RestTemplate restTemplate = new RestTemplate();
+			// the url should be in the format of
+			// "https://server/direct/site/SITE_ID.json"
+			String requestUrl = env.getProperty(Utils.ENV_PROPERTY_CTOOLS_SERVER_URL)
+					+ "direct/content/site/" + siteId + ".json?_sessionId="
+					+ sessionId;
+			String siteResourceJson = null;
+			try {
+				siteResourceJson = restTemplate.getForObject(requestUrl,
+						String.class);
+				 migrationTaskService.boxUploadSiteContent(migrationId, httpContext, remoteUserEmail,
+						sessionId, siteResourceJson, boxFolderId);
+
+			} catch (RestClientException e) {
+				String errorMessage = "Cannot find site by siteId: " + siteId
+						+ " " + e.getMessage();
+				log.error(errorMessage);
+				boxMigrationErrors.append(errorMessage + Utils.LINE_BREAK);
+			} catch (Exception e) {
+				String errorMessage = "Migration status for " + siteId + " "
+						+ e.getClass().getName();
+				log.error("uploadToBox ", e);
+				boxMigrationErrors.append(errorMessage + Utils.LINE_BREAK);
+			}
+
+			String uploadFinished = "Finished upload site content for site "
+					+ siteId;
+			log.info(uploadFinished);
+			rv.put("status", uploadFinished + Utils.LINE_BREAK);
+		} else {
+			String errorBecomeUser = "Problem become user to " + remoteUser;
```

Comment:
Can this be an early return right after the test?  That requires a bit of code duplication (rv.puts and return rv) but INHO would be much clearer.